DimensionalArrayTraits.jl

```diff
@@ -0,0 +1,216 @@
+always happens in smallest to largest order.
+
+Base also defines Forward and Reverse, but they seem overly complicated for our purposes.
+"""
+struct Ordered{D,A} <: Order
+    index::D
+    array::A
+end
+Ordered() = Ordered(Forward(), Forward())
+
+"""
+Trait indicating that the array or dimension has no order.
+"""
+struct Unordered <: Order end
+
+indexorder(order::Ordered) = order.index
+indexorder(order::Unordered) = Unordered()
+arrayorder(order::Ordered) = order.array
+arrayorder(order::Unordered) = Unordered()
+
+"""
+Trait indicating that the array or dimension is in the normal forward order.
+"""
+struct Forward <: Order end
+
+"""
+Trait indicating that the array or dimension is in the reverse order.
+Selector lookup or plotting will be reversed.
+"""
+struct Reverse <: Order end
+
+Base.reverse(::Reverse) = Forward()
+Base.reverse(::Forward) = Reverse()
+Base.reverse(o::Ordered) = Ordered(revese(indexorder(o)), revese(arrayorder(o)))
+Base.reverse(o::Unordered) = Unordered()
+
+
+"""
+Indicates wether the cell value is specific to the locus point
+or is related to the whole the span.
+
+The span may contain a value if the distance between locii if known.
+This will often be identical to the distance between any two sequential
+cell values, but may be distinct due to rounding errors in a vector index,
+or context-dependent spans such as `Month`.
+"""
+abstract type Sampling end
+
+struct Mean <: Sampling end
+
+struct Single <: Sampling end
+
+"""
+Indicate the position of index values in grid cells.
+
+This is frequently `Start` for time series, but may be `Center`
+for spatial data.
+"""
+abstract type Locus end
+
+"""
+Indicates dimensions that are defined by their center coordinates/time/position.
+"""
+struct Center <: Locus end
+
+"""
+Indicates dimensions that are defined by their start coordinates/time/position.
+"""
+struct Start <: Locus end
+
+"""
+Indicates dimensions that are defined by their end coordinates/time/position.
+"""
+struct End <: Locus end
+
+
+"""
+Traits describing the grid type of a dimension.
+"""
+abstract type Grid end
+
+dims(g::Grid) = nothing
+
+"""
+Fallback grid type
+"""
+struct UnknownGrid <: Grid end
+
+
+"""
+Traits describing a grid dimension that is independent of other grid dimensions.
+"""
+abstract type IndependentGrid{O} <: Grid end
+
+abstract type AbstractAllignedGrid{O} <: IndependentGrid{O} end
+
+"""
+Trait describing a grid aligned with a dimension, independent of
+other dimensions.
+
+## Fields
+- `order`: `Order` trait indicating array and index order
+- `locus`: `Locus` trait indicating the position of the indexed point within the cell span
+- `sampling`: `Sampling` trait indicating wether the grid cells are single samples or means
+- `span`: the size of a grid step, such as 1u"km" or `Month(1)`
+"""
+struct AllignedGrid{O<:Order,L<:Locus,Sa<:Sampling,Sp} <: AbstractAllignedGrid{O}
+    order::O
+    locus::L
+    sampling::Sa
+    span::Sp
+end
```

Comment:
Shall this struct contain the actual values of the grid as well? Currently only the step is encoded, without start and end point, which would be needed for a full grid description?